pkg/plugin/plugin.go

```diff
@@ -18,9 +18,49 @@ const (
 	MethodSendNotification = "SendNotification"
 )
 
+// FormElement describe a form element required for the channel config
+type FormElement struct {
```

Comment:
I'd rename this type to something like `ConfigOption`. Yes, the web module will render it as a form element, but that's not all this struct represents. Take the `Name` attribute for example: that's pretty much irrelevant for the form, the only place where it matters is that it will be the key in the JSON object that's stored in the `channel.config` column.

Reply:
`Name` is actually used as the name attribute of form's input elements.

Reply:
> And isn't this the same form like where you select name and type

No. You are mixing up something. This information is stored directly as a property in the type plugin.info. A plugin can only have one type and one name. 

But this `Name` here is used as the name attribute for the input element of the config form. There can be more than one form elements. For example: `address`, `port`, `from` for email.

Reply:
I don't think I'm mixing up something. With form I meant HTML form in the web module. What I wanted to say is the following: currently, this looks something like this (heavily simplified):

```html
<form class="icinga-form icinga-controls" method="POST">
  <input class="primary-submit-btn-duplicate" value="Save Changes" name="submit" type="submit">
  <input value="[...]" name="CSRFToken" required="" type="hidden">
  <input autocomplete="off" aria-required="true" value="[...]" name="name" required="" type="text">
  <select class="autosubmit" disable="" aria-required="true" name="type" required="">[...]</select>
  <input aria-required="true" value="[...]" name="url" required="" type="text">
  <input autocomplete="off" aria-required="true" value="[...]" name="user_id" required="" type="text">
  <input autocomplete="off" aria-required="true" value="[...]" name="token" required="" type="password">
  <input class="btn-remove" formnovalidate="" value="Delete" name="delete" type="submit">
  <input class="btn-primary" value="Save Changes" name="submit" type="submit">
</form>
```

Where the `<input>`/`<select>` elements with `name` being one of `submit`/`CSRFToken`/`name`/`type`/`delete` are generated by Icinga (Notifications) Web. If that was using the `FieldElement.Name` defined here unchanged in the HTML elements, these 5 names (and all others that might be added in the future) would result in a name conflict, so it would be wise for the web to prevent this from the beginning and always use something like `name="config_$name"`.

Reply:
I think I understand your point of view now. You mean, `FormElement.Name` can conflict if the web itself also provide the same name element.
I already handled this problem by using `FieldElement`. 
Web form store these values as following: `values['config'][$name]`